src/components/text-editor/prosemirror-adapter/plugins/menu-state-tracking-plugin.ts

```diff
@@ -7,24 +7,31 @@ import { EditorMenuTypes } from '../menu/types';
 
 export const actionBarPluginKey = new PluginKey('actionBarPlugin');
 
-export type ActiveMenuItems = Record<EditorMenuTypes, boolean>;
+export interface ActiveMenuItems {
+    active: Record<EditorMenuTypes, boolean>;
+    allowed: Record<EditorMenuTypes, boolean>;
+}
 
-export type UpdateMenuItemsCallBack = (activeTypes: ActiveMenuItems) => void;
+export type UpdateMenuItemsCallBack = (
+    activeTypes: Record<EditorMenuTypes, boolean>,
+    allowedTypes: Record<EditorMenuTypes, boolean>,
+) => void;
 
-const getMenuItemStates = (
+export const getMenuItemStates = (
     menuTypes: EditorMenuTypes[],
     menuCommandFactory: MenuCommandFactory,
     view: EditorView,
 ): ActiveMenuItems => {
-    const activeTypes: ActiveMenuItems = {};
+    const activeTypes: Record<EditorMenuTypes, boolean> = {};
+    const allowedTypes: Record<EditorMenuTypes, boolean> = {};
 
     menuTypes.forEach((type) => {
         const command: CommandWithActive = menuCommandFactory.getCommand(type);
-        activeTypes[type] =
-            command && command.active && command.active(view.state);
+        activeTypes[type] = command?.active?.(view.state) || false;
+        allowedTypes[type] = command?.allowed?.(view.state) ?? true;
```

Comment:
```suggestion
        allowedTypes[type] = !!(command?.allowed?.(view.state) ?? true);
```

I think we need the parenthesis in this case, or any potential falsy value would first be converted to `false`, and _then_ compared again `??`, which would prevent the right side of the `??` from ever being executed.

Reply:
⚡ c17bc67664488c4eff60cd9fb4c7bf50cd198554